nectarchain/calibration/container/charge.py

```diff
@@ -0,0 +1,346 @@
+
+        primary_hdu = fits.PrimaryHDU(self.pixels_id,header=hdr)
+        charge_hg_hdu = fits.ImageHDU(self.charge_hg)
+        charge_lg_hdu = fits.ImageHDU(self.charge_lg)
+        peak_hg_hdu = fits.ImageHDU(self.peak_hg)
+        peak_lg_hdu = fits.ImageHDU(self.peak_lg)
+
+        col1 = fits.Column(array = self.event_id, name = "event_id", format = '1I')
+        col2 = fits.Column(array = self.event_type, name = "event_type", format = '1I')
+        col3 = fits.Column(array = self.ucts_timestamp, name = "ucts_timestamp", format = '1K')
+        col4 = fits.Column(array = self.ucts_busy_counter, name = "ucts_busy_counter", format = '1I')
+        col5 = fits.Column(array = self.ucts_event_counter, name = "ucts_event_counter", format = '1I')
+        col6 = fits.Column(array = self.multiplicity, name = "multiplicity", format = '1I')
+
+        coldefs = fits.ColDefs([col1, col2, col3, col4, col5, col6])
+        event_properties = fits.BinTableHDU.from_columns(coldefs)
+
+        col1 = fits.Column(array = self.trig_pattern_all, name = "trig_pattern_all", format = f'{4 * self.CAMERA.n_pixels}L',dim = f'({self.CAMERA.n_pixels},4)')
+        col2 = fits.Column(array = self.trig_pattern, name = "trig_pattern", format = f'{self.CAMERA.n_pixels}L')
+        coldefs = fits.ColDefs([col1, col2])
+        trigger_patern = fits.BinTableHDU.from_columns(coldefs)
+
+        hdul = fits.HDUList([primary_hdu, charge_hg_hdu, charge_lg_hdu,peak_hg_hdu,peak_lg_hdu,event_properties,trigger_patern])
+        try : 
+            hdul.writeto(Path(path)/f"charge_run{self.run_number}{suffix}.fits",overwrite=kwargs.get('overwrite',False))
+            log.info(f"charge saved in {Path(path)}/charge_run{self.run_number}{suffix}.fits")
+        except OSError as e : 
+            log.warning(e)
+        except Exception as e :
+            log.error(e,exc_info = True)
+            raise e
+
+
+
+
+    @staticmethod
+    def from_file(path : Path,run_number : int,**kwargs) : 
+        log.info(f"loading in {path} run number {run_number}")
+        
+        #table = Table.read(Path(path)/f"charge_hg_run{run_number}.ecsv")
+        #pixels_id = table.meta['pixels_id']
+        #charge_hg = np.array([table[colname] for colname in table.colnames]).T
+        #
+        #table = Table.read(Path(path)/f"charge_lg_run{run_number}.ecsv")
+        #charge_lg = np.array([table[colname] for colname in table.colnames]).T
+        #
+        #table = Table.read(Path(path)/f"peak_hg_run{run_number}.ecsv")
+        #peak_hg = np.array([table[colname] for colname in table.colnames]).T
+        #
+        #table = Table.read(Path(path)/f"peak_lg_run{run_number}.ecsv")
+        #peak_lg = np.array([table[colname] for colname in table.colnames]).T
+        #
+        hdul = fits.open(Path(path)/f"charge_run{run_number}.fits")
+        pixels_id = hdul[0].data
```

Comment:
I am note sure here, but in order to prevent issues if the order of extensions in the input FITS files change for one reason or another, to use named extension instead of their index ?

Reply:
I have been looking to astropy documentation and I did not see how to transform the HDUList in a 'HDUDict' to access HDU throw keywords

Reply:
OK, no problem then.